### DISCUSSION TITLE
Parallel testing results

### DISCUSSION THREAD

**Discussion by msl0** [pester/Pester#2563] [General]
I tested this code snippet to run Pester tests in parallel https://gist.github.com/stefanstranger/839044c5c5c146cb97678933e9a001cc. It works fine, but I'm looking for a way to display the combined result in the console. There is a variable $pester which contains the result object, I would like to use it to display a standard test summary. I put more details in the comment under this solution. I'm not so experienced in PowerShell so I would appreciate it if you could give an example

**Comment by fflaten** [comment]
It's not possible to merge them. You're running multiple instances of Pester not aware of eachother.

You could disable output using e.g. `Invoke-Pester -Output None` and create your own custom output in the post processing, but it wouldn't be real-time.

**Reply by msl0** [reply]
I am aware that it would not be real-time
I was looking for a way to reuse existing code from this repo to convert a $pester object containing merged results to an output message format to display a report

**Reply by fflaten** [reply]
The output is generated using a runtime plugin designed to provide a real-time output, so it's not directly compatible with post-processing. The end summary is generated using [Write-PesterReport](https://github.com/pester/Pester/blob/0d3dd6681883f8db2c183a392952e5be383ba1e0/src/functions/Output.ps1#L269) while the text before is scattered around in [Get-WriteScreenPlugin](https://github.com/pester/Pester/blob/0d3dd6681883f8db2c183a392952e5be383ba1e0/src/functions/Output.ps1#L521) if you'd like to reuse some template strings etc.

_Note: `Write-PesterHostMessage` is just a fancy `Write-Host` with ANSI support. You can find/replace all as the parameters are compatible_
